### PR TITLE
[ONLY CODECOV FAILING] Syntax-generator

### PR DESCRIPTION
**Pull request by wxtim**
These changes partially address #3860 

- [x] Made the function update the cylc.lang file.
- [x] Made the function update the cylc.xml file.
- [ ] ~Add function to GH actions~ (moved to follow up)
- [x] Appropriate tests are included (unit and/or functional).
- [x] No dependency changes.

Questions:
- Is this approach sane?
- Can we apply this to the VSCode extension?
- Does this functionality belong within or without this repo?
- Do we want to add output to repo .gitignore to avoid being spammed with change PRs by GH Actions?

### CODE REVIEW

File: cylc/flow/etc/syntax/cylc.lang [hunk 1]
```diff
@@ -86,110 +86,148 @@
...
+        <keyword>abort if timeout handler fails</keyword>
+        <keyword>abort on inactivity</keyword>
+        <keyword>abort on stalled</keyword>
+        <keyword>abort on timeout</keyword>
+        <keyword>aborted handler</keyword>
+        <keyword>batch submit command template</keyword>
+        <keyword>batch system</keyword>
+        <keyword>clock-expire</keyword>
+        <keyword>clock-trigger</keyword>
+        <keyword>collapsed families</keyword>
```
Comment (MetRonnie) [inline cylc/flow/etc/syntax/cylc.lang:105]: I think it's meant to be sorted in reverse, see https://github.com/cylc/cylc-flow/blob/c3ecb20792276227f66d8b375bf1bffdad5143b5/cylc/flow/etc/syntax/cylc.xml#L14

**Review by MetRonnie — COMMENTED** [review]
Looks sensible

**Comment by MetRonnie** [comment]
Probably the main question is: how are we going to incorporate this into our CI? I think having this in "Stage 1: Create release PR" of the GH Actions release process makes the most sense.

> Can we apply this to the VSCode extension?

https://github.com/cylc/cylc-textmate-grammar doesn't currently have any keywords, but it could be possible to use this. We would probably have to output the list of keywords to a JSON or something in `cylc/flow/etc/syntax/` and have a scheduled workflow in cylc-textmate-grammar to checkout cylc-flow and see if the list has changed (like how https://github.com/cylc/Cylc.tmbundle/ checks for updates to cylc-textmate-grammar)

> Does this functionality belong within or without this repo?
> Do we want to add output to repo .gitignore to avoid being spammed with change PRs by GH Actions?

If running as part of the release process, it makes sense to keep in this repo, and we wouldn't have an automated PRs

**Comment by wxtim** [comment]
> Probably the main question is: how are we going to incorporate this into our CI? I think having this in "Stage 1: Create release PR" of the GH Actions release process makes the most sense.

I think @oliver-sanders suggested that this be a follow up issue. I'll create that If you are happy with that @MetRonnie.

**Comment by MetRonnie** [comment]
Can do that in a follow-up issue if you want. Or I could tackle that if you want.

Is this meant to be unmarked as a draft now?

**Review by kinow — APPROVED** [review]

**Comment by oliver-sanders** [comment]
> Probably the main question is: how are we going to incorporate this into our CI?

Very low priority follow on, we can just do it manually for now or add a checkbox to the release process.

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**